pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -287,8 +287,11 @@ func updatePluginList(pluginList interface{}, pluginSet *config.PluginSet, plugi
 
 // QueueSortFunc returns the function to sort pods in scheduling queue
 func (f *framework) QueueSortFunc() LessFunc {
-	if len(f.queueSortPlugins) == 0 {
-		return nil
+	if f == nil || len(f.queueSortPlugins) == 0 {
+		// If framework is nil, or QueueSortPlugin is not registered,
+		// simply keep their order unchanged.
+		// NOTE: this is primarily for tests.
+		return func(_, _ *PodInfo) bool { return false }
```

Comment:
@ahg-g I'm thinking if we really should tolerate `len(f.queueSortPlugins) == 0`, maybe it's better to error this case instead of providing a default implementation? If so, we should either panic here, or throw out the error at:

https://github.com/kubernetes/kubernetes/blob/000888643a5d915041e657677c14a2ff478fefd9/pkg/scheduler/framework/v1alpha1/framework.go#L251-L253

Reply:
sounds good, I think we should do both, error out at line 251 and fatal here.